lazy_theta_star_p_planner/include/lazy_theta_star_p_planner/lazy_theta_star_p.hpp

```diff
@@ -0,0 +1,237 @@
+// Copyright 2020 Anshumaan Singh
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef LAZY_THETA_STAR_P_PLANNER__LAZY_THETA_STAR_P_HPP_
+#define LAZY_THETA_STAR_P_PLANNER__LAZY_THETA_STAR_P_HPP_
+
+#include <cmath>
+#include <vector>
+#include <queue>
+#include <algorithm>
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+
+typedef int id;
+typedef double cost;
+typedef int map_pts;
+typedef double world_pts;
+const double INF_COST = DBL_MAX;
+const int LETHAL_COST = 252;
+
+struct coordsM
+{
+  map_pts x, y;
+};
+
+struct coordsW
+{
+  world_pts x, y;
+};
+
+struct pos
+{
+  id pos_id;
+  cost f;
+};
+
+struct tree_node
+{
+  map_pts x, y;
+  cost g = INF_COST;
+  cost h = INF_COST;
+  id parent_id;
+  bool is_in_queue = false;
+  cost f = INF_COST;
+};
+
+struct comp
+{
+  bool operator()(pos & p1, pos & p2)
+  {
+    return (p1.f) > (p2.f);
+  }
+};
+
+namespace lazyThetaStarP
+{
+class LazyThetaStarP
```

Comment:
Typically we format this with `public` stuff first, with the contructor and destructor. Then functions, then variables